src/Item_Project.php

```diff
@@ -265,16 +269,97 @@ public function getTabNameForItem(CommonGLPI $item, $withtemplate = 0)
+                    self::showForAsset($item);
+                }
+                break;
         }
         return true;
     }
+
+    private static function showForAsset(CommonDBTM $item): void
+    {
+
+        $item_project = new self();
+        $item_projects = $item_project->find([
+            'itemtype' => $item->getType(),
+            'items_id' => $item->getID()
+        ]);
+
+        $used = $entries = [];
+
+        foreach ($item_projects as $value) {
+            $used[] = $value['projects_id'];
+            $project = new Project();
+            $result = $project->getFromDB($value['projects_id']);
+
+            if ($result === false) {
+                continue;
+            }
+
+            $priority = CommonITILObject::getPriorityName($project->fields['priority']);
+            $prioritycolor  = $_SESSION["glpipriority_" . $project->fields['priority']];
+            $state = ProjectState::getById($project->fields['projectstates_id']);
+
+            $data = [
+                'name' => $project->getLink(),
+                'projectstates_id' => $state !== false
+                    ? [
+                        'content' => $state->fields['name'],
+                        'color' => $state->fields['color']
+                    ] : '',
+                'priority' => [
+                    'content' => $priority,
+                    'color' => $prioritycolor
+                ],
+                'percent_done' => Html::getProgressBar((float)$project->fields['percent_done'])
+            ];
+            $entries[] = array_merge($project->fields, $data);
+        }
+
+        $cols = [
+            'columns' => [
+                "name" => __('Name'),
+                "priority" => __('Priority'),
+                "code" => __('Code'),
+                "projectstates_id" => _n('State', 'States', 1),
+                "percent_done" => __('Percent done'),
+                "creation_date" => __('Creation date'),
+                "content" => __('Description'),
+            ],
+            'formatters' => [
+                'name' => 'raw_html',
+                'priority' => 'badge',
+                'projectstates_id' => 'badge',
+                'percent_done' => 'raw_html',
+                'creation_date' => 'date',
+            ]
+        ];
+
+        TemplateRenderer::getInstance()->display('pages/tools/item_project.html.twig', [
+            'item' => $item,
+            'used' => $used
+        ]);
+
+        TemplateRenderer::getInstance()->display('components/datatable.html.twig', [
+            'is_tab' => true,
+            'nopager' => true,
+            'nofilter' => true,
+            'nosort' => true,
+            'columns' => $cols['columns'],
+            'formatters' => $cols['formatters'],
+            'entries' => $entries,
+            'total_number' => count($entries),
+            'filtered_number' => count($entries),
+        ]);
```

Comment:
You could pass all these values in a `datatable_params` variable of your `pages/tools/item_project.html.twig` template and then include the `components/datatable.html.twig` from there.

```twig
{{ include('components/datatable.html.twig', datatable_params, with_context = false) }}
```

It would permit to have render your tab using a unique `TemplateRenderer::display()` call.